relay/sources/relayd/src/output/upstream.rs

```diff
@@ -38,36 +36,39 @@ pub fn send_inventory(
             InventoryType::New => job_config.cfg.output.upstream.default_password.clone(),
             InventoryType::Update => job_config.cfg.output.upstream.password.clone(),
         },
-    ))
+    )
+    .await
 }
 
-fn forward_file(
+async fn forward_file(
     job_config: Arc<JobConfig>,
     endpoint: &str,
     path: PathBuf,
     password: Secret,
-) -> impl Future<Item = (), Error = Error> + '_ {
-    tokio::fs::read(path.clone())
-        .map_err(|e| e.into())
-        .and_then(move |d| {
-            job_config
-                .client
-                .clone()
-                .put(&format!(
-                    "{}/{}/{}",
-                    job_config.cfg.output.upstream.url,
-                    endpoint,
-                    path.file_name().expect("not a file").to_string_lossy()
-                ))
-                .basic_auth(
-                    &job_config.cfg.output.upstream.user,
-                    Some(&password.value()),
-                )
-                .body(d)
-                .send()
-                // HTTP error -> Err()
-                .and_then(|r| r.error_for_status())
-                .map(|r| debug!("Server response: {:#?}", r))
-                .map_err(|e| e.into())
-        })
+) -> Result<(), Error> {
+    let content = tokio::fs::read(path.clone()).await?;
+
+    let result = job_config
+        .client
+        .clone()
+        .put(&format!(
+            "{}/{}/{}",
+            job_config.cfg.output.upstream.url,
+            endpoint,
+            path.file_name().expect("not a file").to_string_lossy()
+        ))
+        .basic_auth(
+            &job_config.cfg.output.upstream.user,
+            Some(&password.value()),
+        )
+        .body(Body::from(content))
+        .send()
+        .await;
+
+    result
```

Comment:
Maybe you'll want to have Err handled here

Reply:
well spotted, fixed!